src/contracts/core/AllocationManager.sol

```diff
@@ -186,50 +222,20 @@ contract AllocationManager is
-                uint64 currentMagnitude = uint64(magnitudeUpdates.upperLookupRecent(uint32(block.timestamp)));
-                // skip if no actively allocated magnitude to slash
-                if (currentMagnitude != 0) {
-                    /// TODO: add wrapping library for rounding up for slashing accounting
-                    slashedMagnitude = uint64(uint256(bipsToSlash) * uint256(currentMagnitude) / BIPS_FACTOR);
-
-                    // propagate slashing to current and future and allocations for the (operator, strategy, operatorSet)
-                    magnitudeUpdates.decrementAtAndFutureSnapshots({
-                        key: uint32(block.timestamp),
-                        decrementValue: slashedMagnitude
-                    });
-                }
-
-            // There can be at most 1 pending allocation + deallocation at a time, hence the else-if statement
-            } else if (numPendingDeallocations != 0) {
-                // Get the number of queued deallocations that can be completed
-                uint256 queuedDeallocationIndicesLen =
-                    _queuedDeallocationIndices[operator][strategies[i]][operatorSetKey].length;
-
-                // index of pendingFreeMagnitude/deallocation to check for slashing
-                uint256 index = _queuedDeallocationIndices[operator][strategies[i]][operatorSetKey][queuedDeallocationIndicesLen - 1];
-                PendingFreeMagnitude storage pendingFreeMagnitude =
-                    _pendingFreeMagnitude[operator][strategies[i]][index];
-
-                // The most recent deallocation may not be completable and, therefore, slashable
-                if (block.timestamp < pendingFreeMagnitude.completableTimestamp) {
-                    // slash the pending deallocation proportionally and store the updated magnitudeDiff
-                    uint64 slashedAmount =
-                        uint64(uint256(bipsToSlash) * uint256(pendingFreeMagnitude.magnitudeDiff) / BIPS_FACTOR);
-                    pendingFreeMagnitude.magnitudeDiff -= slashedAmount;
-                    // keep track of total slashed magnitude
-                    slashedMagnitude += slashedAmount;
-                }
+            // 1. Slash from pending deallocations and allocations
+            MagnitudeInfo memory mInfo = _getCurrentMagnitudeWithPending(operator, strategies[i], operatorSetKey);
+            
+            uint64 slashedMagnitude = uint64(uint256(mInfo.currentMagnitude) * bipsToSlash / BIPS_FACTOR);
+            mInfo.currentMagnitude -= slashedMagnitude;
+            // if there is a pending deallocation, slash pending deallocation proportionally
+            if (mInfo.pendingMagnitudeDiff < 0) {
+                uint256 pendingMagnitudeDiff = uint256(uint128(-mInfo.pendingMagnitudeDiff)) * (BIPS_FACTOR - bipsToSlash) / BIPS_FACTOR;
+                mInfo.pendingMagnitudeDiff += -int128(uint128(pendingMagnitudeDiff));
```

Comment:
This doesn't seam safe, or right. Thinking this can be simplified...

Reply:
Alex pushed a fix here